### DISCUSSION TITLE
Transfer issue from private to public repository?

### DISCUSSION THREAD

**Discussion by britzl** [community/community#21979] [New to GitHub]
<p>We have a couple of issues in a private repository on an organisation that we’d like to move to a public repository on the same organisation. The Transfer Issue dialog does however only show other private repositories. Is this intentional? I do understand that there’s risk involved in moving something from a private to a public repo, but I still believe it should be possible. Is there any way to move an issue thaty I’m not aware of?</p>

**Comment by seveas** [comment, marked as answer]
<p>Hi <a class="mention" href="/britzl">@britzl</a>,</p>
<p>As documented on <a href="https://help.github.com/en/articles/transferring-an-issue-to-another-repository" rel="nofollow noopener">https://help.github.com/en/articles/transferring-an-issue-to-another-repository</a>, it is not possible to transfer an issue from a private repository to a public one.</p>

**Reply by mlissner** [reply]
You can do the transfer with a hack (this is down thread, but posting here for easier finding):
 - Make a private, temporary repo
 - Transfer to the new private repo
 - Make that repo public
 - Transfer the issue to your intended destination
 - Delete the temp repo

If you do this a lot, I suppose the temp repo doesn't have to be temporary. Just name it "transfer-repo" or something, and toggle it back and forth from private to public...

**Comment by britzl** [comment]
<p>Thanks <a class="mention" href="/seveas">@seveas</a>! I can’t believe I overlooked that sentence. Still, I think it should be allowed, with a security warning/confirmation instead of not allowing it at all.</p>

**Comment by seveas** [comment]
<p>Can’t say I disagree, seems like a useful feature. Maybe contact support with a feature request :slight_smile:</p>

**Reply by Goncharo** [reply]
Just did - hoping it is prioritized soon 🙂

**Comment by hotlong** [comment]
<p>+1<br>
I think it should be allowed, with a security warning/confirmation instead of not allowing it at all.</p>

**Comment by rodnymolina** [comment]
<p>+1</p>
<p>It should be allowed with just a warning. The alternative right now is to loose the entire history of the issue and create one in the public repo from scratch. Doesn’t make any sense.</p>

**Comment by idiom-bytes** [comment]
<p>+1</p>
<p>Currently working with a private repo that was used for issue tracking, with many bits being migrated to public repos to improve collaboration and decentralization.</p>
<p>Managed to take the hint after attempting to transfer the issue via gh cli.</p>

**Comment by Erol444** [comment]
<p>+1<br>
It would be really useful to allow this feature, but to warn the user that they will transfer from private to public repo.</p>

**Comment by themarpe** [comment]
<p>+1</p>
<p>To provide a seamless way of moving issues around, if/when needed.</p>

**Comment by tersekmatija** [comment]
<p>+1<br>
That would be totally awesome!</p>

**Comment by hotlong** [comment]
<p>+1<br>
That would be totally awesome!</p>

**Comment by brettpac** [comment]
+1
Yes, this is badly needed.

**Comment by gtk-grafana** [comment]
+1
Agreed, this would save so much time

**Comment by SimonSAMPERE** [comment]
+1

I agree, my organization would also need this feature.

**Comment by evictorero** [comment]
+1
This will be really useful

**Reply by evictorero** [reply]
Well, it can be done with a couple of steps. I hope this is useful for anyone looking how to do this:
* Create a private repository as a placeholder
* Transfer the issues to the placeholder (you can do it in bulk with `gh` CLI)
* Change the placeholder repository to public
* Transfer the issues from the placeholder repository to the public one (again, easier with the CLI)

If you don't want to lose the labels, you must create them first in target repositories.

**Reply by trey** [reply]
Thank you for this tip! That was tedious, but possible. Glad to have it done now! 😄

**Reply by ran-codes** [reply]
it is the year 2023. This tip saved me. thank you

**Comment by Goncharo** [comment]
Still not possible after 4 years of this being open? 😆

**Comment by YetAnotherGerrit** [comment]
+1
This would be awesome!

**Comment by agneszitte** [comment]
+1
At least to give the option to allow it in the settings of the repos if needed (disabled by default)

**Comment by mikermcneil** [comment]
Would love to see this.

**Comment by DiogoLobo122** [comment]
+1 still not avaliable

**Comment by SugarD-x** [comment]
I honestly don't understand why this is not a thing already. GitHub openly promotes making separate public issue repos for private repos that the public can't access because the ability to have public issues in a private repo is still not an option, and yet this is clearly a major block on that "feature" they advertise. I agree with the other users here. Having an option with a warning would be more than sufficient. Especially when this is a much-needed workaround to the request for public issues in private repos, which has had its request on GitHub sitting in limbo for years.

**Comment by Fgerthoffert** [comment]
Same here, we'd really need that option.

For now I have to rely on an action that ends up creating a temporary repository, which ends up being quite tedious and error prone: https://github.com/Fgerthoffert/actions-transfer-issue

On the other hand, a simple API parameter such as "enablePublicRepoTransfer: true" in the [transferIssue mutation](https://docs.github.com/en/graphql/reference/mutations#transferissue) would greatly simplify the job while forcing the user/developer to make a clear decision to perform such a transfer.

**Reply by Fgerthoffert** [reply]
Some of the original instability with [Fgerthoffert/actions-transfer-issue](https://github.com/Fgerthoffert/actions-transfer-issue) was caused by the time needed to transfer the entirety of an issue. 

The transferIssue mutation will respond once the core of the issue has been created, but timelineItems have not yet been fully transferred at that point. And it might take a few seconds for that, so if you end up transferring the issue again before the first operation is complete, this causes some failures. What did the trick for me was to check a bunch of the issue elements (their totalCount) and make sure these were not lower in the transferred issue than the source issue.

Another point of attention, this operation seems to be consuming quite a few of the [secondary rate limit](https://docs.github.com/en/graphql/overview/rate-limits-and-node-limits-for-the-graphql-api#secondary-rate-limits) and while performing the operation in bulk, I did hit errors even though the `x-ratelimit-remaining` still had enough "credits" to perform the operation.

**Comment by slifty** [comment]
The omission of this feature is very silly -- the fact that it is possible to accomplish my goals as a user via a convoluted "transfer repo" makes this restriction particularly pointless.

Just allow this as a primary feature and provide a warning.

**Comment by InsertCreativityHere** [comment]
Just chiming in to also give my
+1
